base/workerpool.jl

```diff
@@ -1,11 +1,26 @@
 # This file is a part of Julia. License is MIT: http://julialang.org/license
 
-type WorkerPool
+abstract AbstractWorkerPool
+
+# An AbstractWorkerPool should implement
+#
+# `push!` - add a new worker into the pool
+# `put!` - put back a worker to the pool
```

Comment:
My penchant for reusing verbs. `take!` and `put!` remove and add back workers from the list of workers available for remote execution - this is maintained in `pool.channel`.

`push!` adds a new worker to the pool itself. Maintained in `pool.workers`